apis/Google.Devtools.Aspnet/Google.Devtools.Aspnet/CloudErrorReportingExceptionLogger.cs

```diff
@@ -0,0 +1,182 @@
+    /// Docs: https://cloud.google.com/error-reporting/docs/
+    /// </remarks>
+    public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
+    {
+        // The formated Google Cloud Platform project id.
+        private readonly string _projectId;
+
+        // The service context in which this error has occurred.
+        // See: https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events#ServiceContext
+        private readonly ServiceContext _serviceContext;
+
+        private readonly ReportErrorsServiceClient _client;
+
+        /// <summary>
+        /// Creates an instance of <see cref="CloudErrorReportingExceptionLogger"/>
+        /// </summary>
+        /// <param name="client">The Error Reporting client.</param>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static CloudErrorReportingExceptionLogger Create(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version)
+        {
+            return new CloudErrorReportingExceptionLogger(client, projectId, serviceName, version);
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="CloudErrorReportingExceptionLogger"/> using credentials as
+        /// defined by <see cref="GoogleCredential.GetApplicationDefaultAsync"/>.
+        /// </summary>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static CloudErrorReportingExceptionLogger Create(string projectId, string serviceName, string version)
+        {
+            return new CloudErrorReportingExceptionLogger(
+                ReportErrorsServiceClient.Create(), projectId, serviceName, version);
+        }
+
+        private CloudErrorReportingExceptionLogger(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version) : base()
+        {
+            _client = GaxPreconditions.CheckNotNull(client, nameof(client));
+            _projectId = ReportErrorsServiceClient.FormatProjectName(
+                GaxPreconditions.CheckNotNull(projectId, nameof(projectId)));
+            _serviceContext = new ServiceContext
+            {
+                Service = GaxPreconditions.CheckNotNull(serviceName, nameof(serviceName)),
+                Version = GaxPreconditions.CheckNotNull(version, nameof(version)),
+            };
+        }
+
+        public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
```

Comment:
Add `/// <inheritdoc />` here and on the other overrides. (It only works in some contexts, but it's at least indicative.)